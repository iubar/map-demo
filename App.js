import React from 'react';
import Map1Screen from './src/screens/Map1Screen';
import Map2Screen from './src/screens/Map2Screen';
import Map3Screen from './src/screens/Map3Screen';
import {
	NavigationContainer,
	DarkTheme as DarkThemeNav,
	DefaultTheme as DefaultThemeNav,
} from '@react-navigation/native';
import MaterialCommunityIcons from 'react-native-vector-icons/MaterialCommunityIcons';

import { createBottomTabNavigator } from '@react-navigation/bottom-tabs';
const Tab = createBottomTabNavigator();

export default class App extends React.Component {
	render() {
		return (
			<NavigationContainer>
				<Tab.Navigator>
					<Tab.Screen
						name="Map 1"
						component={Map1Screen}
						options={{
							tabBarLabel: 'Markers',
							tabBarIcon: ({ color, size }) => (
								<MaterialCommunityIcons name="home" color={color} size={size} />
							),
						}}
					/>
					<Tab.Screen
						name="Map 2"
						component={Map2Screen}
						options={{
							tabBarLabel: 'Events',
							tabBarIcon: ({ color, size }) => (
								<MaterialCommunityIcons name="bell" color={color} size={size} />
							),
							tabBarBadge: 3,
						}}
					/>
					<Tab.Screen
						name="Map 3"
						component={Map3Screen}
						options={{
							tabBarLabel: 'Clusters',
							tabBarIcon: ({ color, size }) => (
								<MaterialCommunityIcons name="account" color={color} size={size} />
							),
						}}
					/>
				</Tab.Navigator>
			</NavigationContainer>
		);
	}
}
