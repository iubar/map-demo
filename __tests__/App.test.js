import React from 'react';
import renderer from 'react-test-renderer';
import App from '../App';
import { act } from 'react-test-renderer';

describe('<App />', () => {
	//it('has 1 child', () => {
	//	const tree = renderer.create(<App />).toJSON();
	//	expect(tree.children.length).toBe(1);
	//});
	it('renders correctly', async () => {
		const tree = renderer.create(<App />).toJSON();

		// see https://github.com/callstack/react-native-testing-library/issues/398
		await act(async () => {
			expect(tree).toMatchSnapshot();
		});
	});
});

/*
describe('<App /> Basics', () => {
  it('has 1 child', async () => {
    let tree
    renderer.act(()=>{
       tree = renderer.create(<App />)
    })
    expect(tree).toMatchSnapshot();
  });
})
*/
