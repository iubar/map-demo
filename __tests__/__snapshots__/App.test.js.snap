// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<App /> renders correctly 1`] = `
<RNCSafeAreaProvider
  onInsetsChange={[Function]}
  style={
    [
      {
        "flex": 1,
      },
      undefined,
    ]
  }
>
  <RNSScreenNavigationContainer
    style={
      {
        "flex": 1,
        "overflow": "hidden",
      }
    }
  >
    <RNSScreen
      activityState={2}
      collapsable={false}
      forwardedRef={[Function]}
      gestureResponseDistance={
        {
          "bottom": -1,
          "end": -1,
          "start": -1,
          "top": -1,
        }
      }
      style={
        {
          "bottom": 0,
          "left": 0,
          "position": "absolute",
          "right": 0,
          "top": 0,
          "zIndex": 0,
        }
      }
    >
      <View
        accessibilityElementsHidden={false}
        importantForAccessibility="auto"
        style={
          [
            {
              "backgroundColor": "rgb(242, 242, 242)",
              "flex": 1,
            },
            [
              {
                "flex": 1,
                "flexDirection": "column-reverse",
              },
              undefined,
            ],
          ]
        }
      >
        <View
          style={
            {
              "flex": 1,
            }
          }
        >
          <View
            style={
              {
                "flex": 1,
              }
            }
          >
            <View
              style={
                {
                  "alignItems": "center",
                  "backgroundColor": "#fff",
                  "flex": 1,
                  "justifyContent": "center",
                }
              }
            >
              <AIRMap
                followsUserLocation={true}
                initialRegion={
                  {
                    "latitude": 37.78825,
                    "latitudeDelta": 0.0922,
                    "longitude": -122.4324,
                    "longitudeDelta": 0.0421,
                  }
                }
                onCalloutPress={[Function]}
                onChange={[Function]}
                onKmlReady={[Function]}
                onLayout={[Function]}
                onMapReady={[Function]}
                onMarkerDeselect={[Function]}
                onMarkerPress={[Function]}
                onMarkerSelect={[Function]}
                onRegionChange={[Function]}
                onRegionChangeComplete={[Function]}
                onUserLocationChange={[Function]}
                region={null}
                showsTraffic={false}
                showsUserLocation={true}
                style={
                  {
                    "height": 1334,
                    "width": 750,
                  }
                }
              >
                <AIRMapMarker
                  coordinate={
                    {
                      "latitude": 37.77825,
                      "longitude": -122.4224,
                    }
                  }
                  description="Description 1"
                  draggable={true}
                  id="0"
                  onDragEnd={[Function]}
                  onPress={[Function]}
                  stopPropagation={false}
                  style={
                    [
                      {
                        "backgroundColor": "transparent",
                        "position": "absolute",
                      },
                      undefined,
                    ]
                  }
                  title="Title 1"
                >
                  <AIRMapCallout
                    alphaHitTest={false}
                    onPress={[Function]}
                    style={
                      [
                        {
                          "position": "absolute",
                        },
                        undefined,
                      ]
                    }
                    tooltip={false}
                  >
                    <View
                      style={
                        {
                          "alignItems": "center",
                          "backgroundColor": "#fff",
                          "flex": 1,
                          "justifyContent": "center",
                        }
                      }
                    >
                      <Text>
                        Title 1
                      </Text>
                      <View
                        accessibilityRole="button"
                        accessible={true}
                        collapsable={false}
                        focusable={true}
                        onClick={[Function]}
                        onResponderGrant={[Function]}
                        onResponderMove={[Function]}
                        onResponderRelease={[Function]}
                        onResponderTerminate={[Function]}
                        onResponderTerminationRequest={[Function]}
                        onStartShouldSetResponder={[Function]}
                        style={
                          {
                            "opacity": 1,
                          }
                        }
                      >
                        <View
                          style={
                            [
                              {},
                            ]
                          }
                        >
                          <Text
                            style={
                              [
                                {
                                  "color": "#007AFF",
                                  "fontSize": 18,
                                  "margin": 8,
                                  "textAlign": "center",
                                },
                                {
                                  "color": "#4169E1",
                                },
                              ]
                            }
                          >
                            Hi World!
                          </Text>
                        </View>
                      </View>
                    </View>
                  </AIRMapCallout>
                </AIRMapMarker>
                <AIRMapMarker
                  coordinate={
                    {
                      "latitude": 37.78825,
                      "longitude": -122.4324,
                    }
                  }
                  description="Description 2"
                  draggable={true}
                  id="1"
                  onDragEnd={[Function]}
                  onPress={[Function]}
                  stopPropagation={false}
                  style={
                    [
                      {
                        "backgroundColor": "transparent",
                        "position": "absolute",
                      },
                      undefined,
                    ]
                  }
                  title="Title 2"
                >
                  <AIRMapCallout
                    alphaHitTest={false}
                    onPress={[Function]}
                    style={
                      [
                        {
                          "position": "absolute",
                        },
                        undefined,
                      ]
                    }
                    tooltip={false}
                  >
                    <View
                      style={
                        {
                          "alignItems": "center",
                          "backgroundColor": "#fff",
                          "flex": 1,
                          "justifyContent": "center",
                        }
                      }
                    >
                      <Text>
                        Title 2
                      </Text>
                      <View
                        accessibilityRole="button"
                        accessible={true}
                        collapsable={false}
                        focusable={true}
                        onClick={[Function]}
                        onResponderGrant={[Function]}
                        onResponderMove={[Function]}
                        onResponderRelease={[Function]}
                        onResponderTerminate={[Function]}
                        onResponderTerminationRequest={[Function]}
                        onStartShouldSetResponder={[Function]}
                        style={
                          {
                            "opacity": 1,
                          }
                        }
                      >
                        <View
                          style={
                            [
                              {},
                            ]
                          }
                        >
                          <Text
                            style={
                              [
                                {
                                  "color": "#007AFF",
                                  "fontSize": 18,
                                  "margin": 8,
                                  "textAlign": "center",
                                },
                                {
                                  "color": "#4169E1",
                                },
                              ]
                            }
                          >
                            Hi World!
                          </Text>
                        </View>
                      </View>
                    </View>
                  </AIRMapCallout>
                </AIRMapMarker>
                <AIRMapMarker
                  coordinate={
                    {
                      "latitude": 37.79825,
                      "longitude": -122.4424,
                    }
                  }
                  description="Description 3"
                  draggable={true}
                  id="2"
                  onDragEnd={[Function]}
                  onPress={[Function]}
                  stopPropagation={false}
                  style={
                    [
                      {
                        "backgroundColor": "transparent",
                        "position": "absolute",
                      },
                      undefined,
                    ]
                  }
                  title="Title 3"
                >
                  <AIRMapCallout
                    alphaHitTest={false}
                    onPress={[Function]}
                    style={
                      [
                        {
                          "position": "absolute",
                        },
                        undefined,
                      ]
                    }
                    tooltip={false}
                  >
                    <View
                      style={
                        {
                          "alignItems": "center",
                          "backgroundColor": "#fff",
                          "flex": 1,
                          "justifyContent": "center",
                        }
                      }
                    >
                      <Text>
                        Title 3
                      </Text>
                      <View
                        accessibilityRole="button"
                        accessible={true}
                        collapsable={false}
                        focusable={true}
                        onClick={[Function]}
                        onResponderGrant={[Function]}
                        onResponderMove={[Function]}
                        onResponderRelease={[Function]}
                        onResponderTerminate={[Function]}
                        onResponderTerminationRequest={[Function]}
                        onStartShouldSetResponder={[Function]}
                        style={
                          {
                            "opacity": 1,
                          }
                        }
                      >
                        <View
                          style={
                            [
                              {},
                            ]
                          }
                        >
                          <Text
                            style={
                              [
                                {
                                  "color": "#007AFF",
                                  "fontSize": 18,
                                  "margin": 8,
                                  "textAlign": "center",
                                },
                                {
                                  "color": "#4169E1",
                                },
                              ]
                            }
                          >
                            Hi World!
                          </Text>
                        </View>
                      </View>
                    </View>
                  </AIRMapCallout>
                </AIRMapMarker>
              </AIRMap>
              <View
                accessible={true}
                collapsable={false}
                focusable={false}
                onClick={[Function]}
                onResponderGrant={[Function]}
                onResponderMove={[Function]}
                onResponderRelease={[Function]}
                onResponderTerminate={[Function]}
                onResponderTerminationRequest={[Function]}
                onStartShouldSetResponder={[Function]}
                style={
                  {
                    "alignItems": "center",
                    "backgroundColor": "#fff",
                    "borderColor": "#fff",
                    "borderRadius": 10,
                    "borderWidth": 1,
                    "bottom": 50,
                    "justifyContent": "center",
                    "opacity": 0.5,
                    "paddingBottom": 10,
                    "paddingLeft": 10,
                    "paddingRight": 10,
                    "paddingTop": 10,
                    "position": "absolute",
                  }
                }
              >
                <Text>
                  Lat: 37.78825
                </Text>
                <Text>
                  Lon: -122.4324
                </Text>
              </View>
            </View>
            <View
              style={
                {
                  "alignItems": "center",
                  "backgroundColor": "black",
                  "height": 100,
                  "justifyContent": "center",
                }
              }
            >
              <Text
                style={
                  {
                    "color": "white",
                  }
                }
              >
                Hello !
              </Text>
            </View>
          </View>
        </View>
        <View
          onLayout={[Function]}
          style={null}
        >
          <View
            collapsable={false}
            pointerEvents="box-none"
            style={
              {
                "bottom": 0,
                "left": 0,
                "position": "absolute",
                "right": 0,
                "top": 0,
                "zIndex": 0,
              }
            }
          >
            <View
              collapsable={false}
              style={
                {
                  "backgroundColor": "rgb(255, 255, 255)",
                  "borderBottomColor": "rgb(216, 216, 216)",
                  "flex": 1,
                  "shadowColor": "rgb(216, 216, 216)",
                  "shadowOffset": {
                    "height": 0.5,
                    "width": 0,
                  },
                  "shadowOpacity": 0.85,
                  "shadowRadius": 0,
                }
              }
            />
          </View>
          <View
            collapsable={false}
            pointerEvents="box-none"
            style={
              {
                "height": 44,
                "maxHeight": undefined,
                "minHeight": undefined,
                "opacity": undefined,
                "transform": undefined,
              }
            }
          >
            <View
              pointerEvents="none"
              style={
                {
                  "height": 0,
                }
              }
            />
            <View
              pointerEvents="box-none"
              style={
                {
                  "alignItems": "stretch",
                  "flex": 1,
                  "flexDirection": "row",
                }
              }
            >
              <View
                collapsable={false}
                pointerEvents="box-none"
                style={
                  {
                    "alignItems": "flex-start",
                    "flexBasis": 0,
                    "flexGrow": 1,
                    "justifyContent": "center",
                    "marginStart": 0,
                  }
                }
              />
              <View
                collapsable={false}
                pointerEvents="box-none"
                style={
                  {
                    "justifyContent": "center",
                    "marginHorizontal": 16,
                    "maxWidth": 718,
                  }
                }
              >
                <Text
                  accessibilityRole="header"
                  aria-level="1"
                  collapsable={false}
                  numberOfLines={1}
                  style={
                    {
                      "color": "rgb(28, 28, 30)",
                      "fontSize": 17,
                      "fontWeight": "600",
                    }
                  }
                >
                  Map1
                </Text>
              </View>
              <View
                collapsable={false}
                pointerEvents="box-none"
                style={
                  {
                    "alignItems": "flex-end",
                    "flexBasis": 0,
                    "flexGrow": 1,
                    "justifyContent": "center",
                    "marginEnd": 0,
                  }
                }
              />
            </View>
          </View>
        </View>
      </View>
    </RNSScreen>
  </RNSScreenNavigationContainer>
  <View
    collapsable={false}
    onLayout={[Function]}
    pointerEvents="auto"
    style={
      {
        "backgroundColor": "rgb(255, 255, 255)",
        "borderTopColor": "rgb(216, 216, 216)",
        "borderTopWidth": 0.5,
        "bottom": 0,
        "elevation": 8,
        "height": 49,
        "left": 0,
        "paddingBottom": 0,
        "paddingHorizontal": 0,
        "position": null,
        "right": 0,
        "transform": [
          {
            "translateY": 0,
          },
        ],
      }
    }
  >
    <View
      pointerEvents="none"
      style={
        {
          "bottom": 0,
          "left": 0,
          "position": "absolute",
          "right": 0,
          "top": 0,
        }
      }
    />
    <View
      accessibilityRole="tablist"
      style={
        {
          "flex": 1,
          "flexDirection": "row",
        }
      }
    >
      <View
        accessibilityLabel="Markers, tab, 1 of 3"
        accessibilityRole="button"
        accessibilityState={
          {
            "selected": true,
          }
        }
        accessibilityStates={
          [
            "selected",
          ]
        }
        accessible={true}
        collapsable={false}
        focusable={true}
        onBlur={[Function]}
        onClick={[Function]}
        onFocus={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          [
            {
              "alignItems": "center",
              "flex": 1,
            },
            {
              "backgroundColor": "transparent",
            },
            {
              "flexDirection": "column",
              "justifyContent": "flex-end",
            },
            undefined,
          ]
        }
      >
        <View
          style={
            [
              {
                "flex": 1,
              },
              undefined,
            ]
          }
        >
          <View
            style={
              [
                {
                  "alignItems": "center",
                  "alignSelf": "center",
                  "height": "100%",
                  "justifyContent": "center",
                  "minWidth": 25,
                  "position": "absolute",
                  "width": "100%",
                },
                {
                  "opacity": 1,
                },
              ]
            }
          >
            <Text />
          </View>
          <View
            style={
              [
                {
                  "alignItems": "center",
                  "alignSelf": "center",
                  "height": "100%",
                  "justifyContent": "center",
                  "minWidth": 25,
                  "position": "absolute",
                  "width": "100%",
                },
                {
                  "opacity": 0,
                },
              ]
            }
          >
            <Text />
          </View>
        </View>
        <Text
          numberOfLines={1}
          style={
            [
              {
                "backgroundColor": "transparent",
                "textAlign": "center",
              },
              {
                "color": "rgb(0, 122, 255)",
              },
              {
                "fontSize": 10,
              },
              undefined,
            ]
          }
        >
          Markers
        </Text>
      </View>
      <View
        accessibilityLabel="Events, tab, 2 of 3"
        accessibilityRole="button"
        accessibilityState={
          {
            "selected": false,
          }
        }
        accessibilityStates={[]}
        accessible={true}
        collapsable={false}
        focusable={true}
        onBlur={[Function]}
        onClick={[Function]}
        onFocus={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          [
            {
              "alignItems": "center",
              "flex": 1,
            },
            {
              "backgroundColor": "transparent",
            },
            {
              "flexDirection": "column",
              "justifyContent": "flex-end",
            },
            undefined,
          ]
        }
      >
        <View
          style={
            [
              {
                "flex": 1,
              },
              undefined,
            ]
          }
        >
          <View
            style={
              [
                {
                  "alignItems": "center",
                  "alignSelf": "center",
                  "height": "100%",
                  "justifyContent": "center",
                  "minWidth": 25,
                  "position": "absolute",
                  "width": "100%",
                },
                {
                  "opacity": 0,
                },
              ]
            }
          >
            <Text />
          </View>
          <View
            style={
              [
                {
                  "alignItems": "center",
                  "alignSelf": "center",
                  "height": "100%",
                  "justifyContent": "center",
                  "minWidth": 25,
                  "position": "absolute",
                  "width": "100%",
                },
                {
                  "opacity": 1,
                },
              ]
            }
          >
            <Text />
          </View>
          <Text
            collapsable={false}
            numberOfLines={1}
            style={
              {
                "alignSelf": "flex-end",
                "backgroundColor": "rgb(255, 59, 48)",
                "borderRadius": 9.375,
                "color": "white",
                "fontSize": 14,
                "height": 18.75,
                "left": 3,
                "lineHeight": 17.75,
                "minWidth": 18.75,
                "opacity": 1,
                "overflow": "hidden",
                "paddingHorizontal": 4,
                "position": "absolute",
                "textAlign": "center",
                "top": 3,
                "transform": [
                  {
                    "scale": 1,
                  },
                ],
              }
            }
          >
            3
          </Text>
        </View>
        <Text
          numberOfLines={1}
          style={
            [
              {
                "backgroundColor": "transparent",
                "textAlign": "center",
              },
              {
                "color": "#8E8E8F",
              },
              {
                "fontSize": 10,
              },
              undefined,
            ]
          }
        >
          Events
        </Text>
      </View>
      <View
        accessibilityLabel="Clusters, tab, 3 of 3"
        accessibilityRole="button"
        accessibilityState={
          {
            "selected": false,
          }
        }
        accessibilityStates={[]}
        accessible={true}
        collapsable={false}
        focusable={true}
        onBlur={[Function]}
        onClick={[Function]}
        onFocus={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          [
            {
              "alignItems": "center",
              "flex": 1,
            },
            {
              "backgroundColor": "transparent",
            },
            {
              "flexDirection": "column",
              "justifyContent": "flex-end",
            },
            undefined,
          ]
        }
      >
        <View
          style={
            [
              {
                "flex": 1,
              },
              undefined,
            ]
          }
        >
          <View
            style={
              [
                {
                  "alignItems": "center",
                  "alignSelf": "center",
                  "height": "100%",
                  "justifyContent": "center",
                  "minWidth": 25,
                  "position": "absolute",
                  "width": "100%",
                },
                {
                  "opacity": 0,
                },
              ]
            }
          >
            <Text />
          </View>
          <View
            style={
              [
                {
                  "alignItems": "center",
                  "alignSelf": "center",
                  "height": "100%",
                  "justifyContent": "center",
                  "minWidth": 25,
                  "position": "absolute",
                  "width": "100%",
                },
                {
                  "opacity": 1,
                },
              ]
            }
          >
            <Text />
          </View>
        </View>
        <Text
          numberOfLines={1}
          style={
            [
              {
                "backgroundColor": "transparent",
                "textAlign": "center",
              },
              {
                "color": "#8E8E8F",
              },
              {
                "fontSize": 10,
              },
              undefined,
            ]
          }
        >
          Clusters
        </Text>
      </View>
    </View>
  </View>
</RNCSafeAreaProvider>
`;
